api/api/utils/photon.py

```diff
@@ -63,7 +65,7 @@ def check_image_type(image_url: str, media_obj) -> None:
     if not ext:
         # If the extension is still not present, try getting it from the content type
         try:
-            response = requests.head(image_url)
+            response = requests.head(image_url, timeout=10)
```

Comment:
#2426 is already merged, so this change isn't necessary here (rebase should fix it).